src/interpreter/kernel/exec.js

```diff
@@ -921,15 +968,22 @@ export function executeStackFrame(
     }
 
     if (typeof frame.trace === "function") {
-      frame.trace(depth, pc, instruction);
+      frame.trace(depth, frame._pc, instruction, frame);
     }
 
-    pc++;
+    frame._pc++;
   }
 
   // Return the item on top of the values/stack;
   if (frame.values.length > 0) {
-    return pop1();
+    const res = pop1();
+
+    if (res.type !== "label") {
+      return res;
+    } else {
+      // Push label back
+      pushResult(res);
+    }
```

Comment:
That's a problem with the current working of the interpreter, we should end up with a label here.